gradle/libs.versions.toml

```
@@ -0,0 +1,27 @@
+[versions]
+# this is our own version we deploy
+baseVersion =  "1.29.2"
+
+#versions
+kotlinApi = "1.7"
+kotlin = "1.7.10"
+kotlinJVMtarget = "1.8"
+
+# plugins
+kotlinCompatibilityValidator = "0.13.2"
+
+# libraries
+semver = "1.1.2"
+junit = "4.13.2"
+
+git-based-versioning = "*"
+buildBackends = "2.4.0+"
+
+[plugins]
+kotlin-compatibility-validator = { id = "org.jetbrains.kotlinx.binary-compatibility-validator", version.ref = "kotlinCompatibilityValidator" }
```

I'd argue that extracting versions that are used only once into `[versions]` brings no real benefits, you can put the version directly here
```suggestion
kotlin-compatibility-validator = "org.jetbrains.kotlinx.binary-compatibility-validator:0.13.2"
```

---

I think there are several benefits to also pull single mentioned versions into the TOML, taken from the Gradle docs:

- Type-Safe Accessors
- Centralized Version Management
- Version Separation

But most importantly machine readable versions for renovate or other bots are a core benefit. I think this adds value and would strongly vote to keep it.

---

Renovate can read embedded versions, and we don’t need to access versions in a type-safe manner. I don’t understand the benefit of version separation here either. To the contrary, things that belong together should stay together.

---

I guess it depends whether or not we agree on using the version catalog? I am fine in dropping this PR ifthere is general doubt from your perspective...

But if we want to use a version catalog then leaving this single version out would be the actual separation from my perspective.

---

I don't mean this single version should be left out but all versions that are only used from one place (e.g. semver, too).

In my view, extracting the versions into [versions] section makes sense for versions of related libraries built from one repo, such as Kotlin or JUnit 5 (which has `org.junit.jupiter:junit-jupiter-api`, `org.junit.jupiter:junit-jupiter-engine`, `org.junit.jupiter:junit-jupiter-params` etc. all using the same version).

I agree that it makes sense to extract all dependencies into the version catalog to avoid projects accidentally using different and slightly incompatible versions. But I don't think it makes sense to extract _all_ versions into the [versions] section, only where it helps avoid duplication.

---

Also, I wouldn't expect to find _our own_ version in libs.versions.toml, that's after all for libraries, i.e. dependencies.

---

After async discussion we concluded that we will add all versions to the catalog but leave out the own version.